> Do you have a reproducible test case (dockerfile or something) for the thing not compiling?

Ah, found it. It does not work with `clang` as compiler but works fine with `gcc`.
```
FROM alpine:3.15
RUN apk add make git clang musl-dev gcc

RUN git clone https://github.com/ocaml/ocaml -b 4.14
WORKDIR ocaml/

# Uncomment this to make it work
# RUN sed -i -e "s/amd64|s390x/amd64|arm64|s390x/" configure

RUN ./configure CC="clang" AS="clang -c" ASPP="clang -c"
RUN make -j$(nproc)
```
```
Step 6/6 : RUN make -j$(nproc)
 ---> Running in 983a211eb0b7
make -C runtime sak
make[1]: Entering directory '/ocaml/runtime'
mkdir -p .dep
tr -d '\r' < caml/instruct.h | \
sed -e '/\/\*/d' \
    -e '/^#/d' \
    -e 's/enum /static char * names_of_/' \
    -e 's/{$/[] = {/' \
    -e 's/\([[:upper:]][[:upper:]_0-9]*\)/"\1"/g' > caml/opnames.h
tr -d '\r' < caml/instruct.h | \
sed -n -e '/^  /s/ \([A-Z]\)/ \&\&lbl_\1/gp' \
       -e '/^}/q' > caml/jumptbl.h
clang -c -no-pie -O2 -fno-strict-aliasing -fwrapv -pthread -Wall -Wdeclaration-after-statement -Werror -fno-common -ffunction-sections -g  -D_FILE_OFFSET_BITS=64 -DCAML_NAME_SPACE  -DCAMLDLLIMPORT= -DNATIVE_CODE -DTARGET_arm64 -DMODEL_default -DSYS_linux  -o sak.o sak.c
clang-12: error: argument unused during compilation: '-no-pie' [-Werror,-Wunused-command-line-argument]
make[1]: Leaving directory '/ocaml/runtime'
make[1]: *** [Makefile:256: sak.o] Error 1
make: *** [Makefile:785: runtime/sak] Error 2
The command '/bin/sh -c make -j$(nproc)' returned a non-zero code: 2
```

--- a/configure
+++ b/configure
@@ -14477,7 +14477,7 @@ fi
 # e.g. Ubuntu >= 17.10 for i386, which uses the glibc dynamic loader.
 
 case $arch in #(
-  amd64|aarch64|arm64|s390x|none) :
+  amd64|arm64|s390x|none) :
     # ocamlopt generates PIC code or doesn't generate code at all
      ;; #(
   *) :

--- a/configure.ac
+++ b/configure.ac
@@ -1181,7 +1181,7 @@ AS_IF([test -z "$PARTIALLD"],
 # e.g. Ubuntu >= 17.10 for i386, which uses the glibc dynamic loader.
 
 AS_CASE([$arch],
-  [amd64|aarch64|arm64|s390x|none],
+  [amd64|arm64|s390x|none],
     # ocamlopt generates PIC code or doesn't generate code at all
     [],
   [AS_CASE([$host],

